tuxemon/states/control/__init__.py

```diff
@@ -0,0 +1,129 @@
+#
+# Tuxemon
+# Copyright (C) 2014, William Edwards <shadowapex@gmail.com>,
+#                     Benjamin Bean <superman2k5@gmail.com>
+#
+# This file is part of Tuxemon.
+#
+# Tuxemon is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Tuxemon is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Tuxemon.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Contributor(s):
+#
+# Andrew Hong <novialriptide@gmail.com>
+#
+#
+# states.start Handles the start screen which loads and creates new games
+#
+"""This module contains the Options state
+"""
+from __future__ import annotations
+\
+from functools import partial
+from typing import Any, Callable, Union, Optional
+
+import pygame
+
+from tuxemon import prepare, config
+from tuxemon.constants import paths
+from tuxemon.locale import T
+from tuxemon.menu.interface import MenuItem
+from tuxemon.menu.menu import PopUpMenu
+from tuxemon.platform.events import PlayerInput
+from tuxemon.state import State
+
+ControlStateObj = Callable[[], object]
+cfg_file = config.TuxemonConfig(paths.USER_CONFIG_PATH).cfg
+config = prepare.CONFIG
+
+def reload_control_config():
+    config.a = cfg_file.get("controls", "a")
+    config.b = cfg_file.get("controls", "b")
+    config.up = cfg_file.get("controls", "up")
+    config.left = cfg_file.get("controls", "left")
+    config.down = cfg_file.get("controls", "down")
+    config.right = cfg_file.get("controls", "right")
+    config.back = cfg_file.get("controls", "back")
+
+class SetKeyState(PopUpMenu):
+    """This state is responsible for setting the input keys"""
+    shrink_to_items = True
+
+    def startup(self, **kwargs: Any) -> None:
+        self.input = kwargs["input"]
+        super().startup(**kwargs)
+
+        menu_items_map = (
+            T.translate("options_new_input_key0"),
+            T.translate("options_new_input_key1")
+        )
+
+        for key in menu_items_map:
+            label = T.translate(key).upper()
+            image = self.shadow_text(label)
+            item = MenuItem(image, label, None, None)
+            self.add(item)
+
+    def process_event(self, event: PlayerInput) -> Optional[PlayerInput]:
```

Comment:
please add docstrings like this for this and the other new functions/methods added in the pr
```
"""
Some short description

Parameters:
    event: Player Input to be checked

"""
```

Reply:
This is an inherited method. The docstrings are automatically inherited, so I would only override them if there is something to change/add.

Reply:
good point, thank you